core/app/models/spree/stock_location.rb

```diff
@@ -99,6 +101,16 @@ def fill_status(variant, quantity)
       end
     end
 
+    def self.default
+      Rails.cache.fetch("#{Rails.application.class.parent_name.underscore}_default_stock_location") do
+        default_location = where(default: true).first
+        if default_location.blank? && (default_location = first)
+          default_location.update(default: true)
```

Comment:
In case this update fails for validation reason the error gets shadowed unless you use `update!`. I suggest to use the bang method when you do not check the operation explicitly.

I do not think this "silent data migration" belongs here at all, but if so it should not silently fail.